dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaServicesTask.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager;
+
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.MetadataDumperUsageException;
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.ClouderaManagerHandle.ClouderaClusterDTO;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+
+public class ClouderaServicesTask extends AbstractClouderaManagerTask {
+
+  public ClouderaServicesTask() {
+    super("services.jsonl");
+  }
+
+  @Override
+  protected Void doRun(
+      TaskRunContext context, @Nonnull ByteSink sink, @Nonnull ClouderaManagerHandle handle)
+      throws Exception {
+    CloseableHttpClient httpClient = handle.getHttpClient();
+    List<ClouderaClusterDTO> clusters = handle.getClusters();
+    if (clusters == null) {
+      throw new MetadataDumperUsageException(
+          "Cloudera clusters must be initialized before services dumping.");
+    }
+
+    try (Writer writer = sink.asCharSink(StandardCharsets.UTF_8).openBufferedStream()) {
+      for (ClouderaClusterDTO cluster : clusters) {
+        String servicesPerCluster =
+            handle.getApiURI() + "/clusters/" + cluster.getName() + "/services";
+
+        try (CloseableHttpResponse services = httpClient.execute(new HttpGet(servicesPerCluster))) {
+          String json = EntityUtils.toString(services.getEntity());
+          writer.write(jsonToJsonl(json));
+          writer.write('\n');
+        }
```

Comment:
I'd recommend to write all errors to a separate file - similar to exceptions.txt produced by the dumper for jdbc connectors.
This way it will be easier to debug problems like not all services present in the dump. (That can be done in a separate PR)

Reply:
This applies to other tasks too.

Reply:
As discussed with @dominik-klimczak it should work out of the box.